examples/pytorch/graphsage/experimental/train_dist.py

```diff
@@ -184,7 +184,8 @@ def pad_data(nids, device):
 def run(args, device, data):
     # Unpack data
     train_nid, val_nid, test_nid, in_feats, n_classes, g = data
-    train_nid = pad_data(train_nid, device)
+    # Below line causes error sometimes.
+    # train_nid = pad_data(train_nid, device)
```

Comment:
what does it do?

Reply:
This is a fix when batch size is small. The nid might be unequal between trainers, and when batch size is small, the number of batches will be unequal across trainer, which results in hanging for the trainer with more batches. 
Here I pad the nid to the same length (as the max one across trainer), by repeating the starting ids. But here cannot handle duplicate node ids within the same batches, thus might cause error when shuffling.

Now we decided to adopt drop last strategy, that drop the batch for the trainer with more batches than others

Reply:
This is a fix when batch size is small. The nid might be unequal between trainers, and when batch size is small, the number of batches will be unequal across trainer, which results in hanging for the trainer with more batches. 
Here I pad the nid to the same length (as the max one across trainer), by repeating the starting ids. But here cannot handle duplicate node ids within the same batches, thus might cause error when shuffling.

Now we decided to adopt drop last strategy, that drop the batch for the trainer with more batches than others

Reply:
Issue: https://github.com/dmlc/dgl/issues/3538